### DISCUSSION TITLE
Flashing only certain artifacts into Jetson Xavier devices (not the complete image)

### DISCUSSION THREAD

**Discussion by armhzjz** [OE4T/meta-tegra#602] [Q&A] [answered]
I managed to boot from tftp and to mount the rootfs from NFS.
But I noticed that if something goes wrong during booting the device will either boot or mount the rootfs from eMMC.

Since I haven't found any documentation on how to properly use the tegraflash tooling generated by the build (and its capabilities); does anybody know if it is possible to flash ___only___ the bootloader into the device? 
So that if for any reason, booting from tftp is not possible the booting will fail instead of resorting to booting from eMMC (my device does this even if I explicitly rule out all the possible boot sources except `net` on the CBoot dtb.

**Comment by madisongh** [comment, marked as answer]
You should be able to create a flash layout file that omits the kernel, APP, and any other partitions not needed for running the bootloader. You could also try patching cboot to prevent it from trying other boot devices, too.

As far as flashing specific partitions - the flash helper script has a `-c` option for passing a specific command to tegraflash, rather than than the default `flash;reboot` string; so specifying `-c "write <PARTNAME> <FILENAME>"` to write a single partition should work. However, the boot chain requires flashing several partitions. It would probably be simpler to just create a custom flash layout and/or cboot for your particular use case than to figure out exactly which partitions and files need to be flashed.

For Xavier NX devices (and the SDcard-based Nano development kits), there is a `--spi-only` option you can specify, which causes the flash helper script to filter out the eMMC/SDcard partitions so only the SPI flash-based partitions get flashed. That probably won't do what you need, though, since there are *some* partitions in the eMMC on the Xavier NX that would need to get flashed, too.

**Comment by armhzjz** [comment]
Thanks again for your support @madisongh
